### PR TITLE
drop empty buffers

### PR DESCRIPTION
**Pull request by havardpe**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@baldersheim please review
@arnej27959 FYI

### CODE REVIEW

File: vespalib/src/vespa/vespalib/net/tls/maybe_tls_crypto_socket.cpp [hunk 1]
```diff
@@ -77,12 +77,14 @@ class MyCryptoSocket : public CryptoSocket
         if (frame > 0) {
             memcpy(buf, src.data, frame);
             _buffer.evict(frame);
+            _buffer.drop_if_empty();
```
Comment (arnej27959) [inline vespalib/src/vespa/vespalib/net/tls/maybe_tls_crypto_socket.cpp:80]: should not this `drop_if_empty` be in `drop_empty_buffers()` below?

**Review by baldersheim — APPROVED** [review]
Looks good so far. Note the comment from @arnej27959

File: vespalib/src/vespa/vespalib/net/tls/maybe_tls_crypto_socket.cpp (hunk 1 above)
Comment (havardpe) [reply vespalib/src/vespa/vespalib/net/tls/maybe_tls_crypto_socket.cpp:80]: Actually, no. This is to discard the TLS snoop buffer after it has been drained in the non-tls case. We do direct read/write to the socket in this mode, and do not need the 4_Ki buffer used for snooping during handshaking.

**Comment by baldersheim** [comment]
Wrong button ? close != merge

**Comment by havardpe** [comment]
I got a merge conflict when I tried to merge, then I rebased on top of master and did a force push, which gave me a 0/0 branch since it was already merged. Github API has signed the merge commit, making it 'verified', but will not take responsibility for it and claims it was merged in some other way or commited directly to the master branch...

**Pull request closed without merging**